meadow-client/src/Gists.purs

```diff
@@ -1,163 +1,211 @@
+        ]
+    Success GithubUser -> authorisedView
+    Loading ->
+      button
+        [ idPublishGist
+        , classes [btn, btnInfo, pullRight]
+        , disabled true
+        ] [icon Spinner]
+    NotAsked ->
+      button
+        [ idPublishGist
+        , classes [btn, btnInfo, pullRight]
+        , disabled true
+        ] [icon Spinner]
+
+  publishButton = case createGistResult of
+    Failure _ ->
+      button
+        [ idPublishGist
+        , classes [btn, btnBlock, btnSmall, btnDanger]
+        ] [text "Failure"]
+    Success _ ->
+      button
+        [ idPublishGist
+        , classes [btn, btnBlock, btnSmall, btnPrimary]
+        , onClick $ input_ PublishGist
+        ] [icon Github, nbsp, text "Republish"]
+    Loading ->
+      button
+        [ idPublishGist
+        , classes [btn, btnBlock, btnSmall, btnInfo]
+        , disabled true
+        ] [icon Spinner]
+    NotAsked ->
+      button
+        [ idPublishGist
+        , classes [btn, btnBlock, btnSmall, btnPrimary]
+        , onClick $ input_ PublishGist
+        ] [icon Github, nbsp, text "Publish"]
+
+  loadMessage = [icon Github, nbsp, text "Load"]
+
+  loadButton = case createGistResult of
+    Loading -> empty
+    _ ->
+      button
+        [ idLoadGist
+        , classes
+            [ btn
+            , btnBlock
+            , btnSmall
+            , case parsedGistId of
+                Just (Left url) -> btnDanger
+                Just (Right url) -> btnPrimary
+                Nothing -> btnInfo
+            ]
+        , onClick $ input_ LoadGist
+        , disabled
+            $ case parsedGistId of
+                Just (Left url) -> true
+                Just (Right url) -> false
+                Nothing -> true
+        ] loadMessage
 
-gistPane ::
-  forall p i.
-  Gist ->
-  HTML p i
+gistPane :: forall p i. Gist -> HTML p i
 gistPane gist =
-  div
-    [ class_ $ ClassName "gist-link"
-    ]
+  div [class_ $ ClassName "gist-link"]
     [ a
         [ href $ view gistHtmlUrl gist
         , target "_blank"
-        ]
-        [ text $ "View on Github"
-        ]
+        ] [text $ "View on Github"]
     ]
 
 mkNewGist ::
-  Maybe String ->
+  Maybe SourceCode ->
   Maybe NewGist
-mkNewGist mSource = if Array.null gistFiles
+mkNewGist source = if Array.null gistFiles
   then Nothing
   else
     Just
       $ NewGist
-          { _newGistDescription: "Plutus Playground Smart Contract"
+          { _newGistDescription: "Marlowe Smart Contract"
           , _newGistPublic: true
           , _newGistFiles: gistFiles
           }
   where
   gistFiles =
     catMaybes
-      [ mkNewGistFile "Playground.hs" <$> mSource
+      [ mkNewGistFile gistSourceFilename <<< unwrap <$> source
       ]
 
   mkNewGistFile _newGistFilename _newGistFileContent =
     NewGistFile
       { _newGistFilename
       , _newGistFileContent
       }
+
+gistSourceFilename :: String
+gistSourceFilename = "Meadow.hs"
+
+gistIdInLinkRegex :: Either String Regex
```

Comment:
Can we move from here down into `web-common`? Seems an easy place to start fighting the copy-paste divergence.

Reply:
done